client/state/08types.go

```diff
@@ -56,22 +76,38 @@ func (t *taskRunnerState08) Upgrade() *state.LocalState {
 		PrestartDone: t.TaskDirBuilt,
 	}
 
+	// Don't need logmon in pre09 tasks
+	ls.Hooks["logmon"] = &state.HookState{
+		PrestartDone: true,
+	}
+
 	// Upgrade dispatch payload state
 	ls.Hooks["dispatch_payload"] = &state.HookState{
 		PrestartDone: t.PayloadRendered,
 	}
 
-	//TODO How to convert handles?! This does not work.
-	ls.TaskHandle = drivers.NewTaskHandle("TODO")
-
-	//TODO where do we get this from?
-	ls.TaskHandle.Config = nil
+	// Add nessicary fields to TaskConfig
+	ls.TaskHandle = drivers.NewTaskHandle(0)
+	ls.TaskHandle.Config = &drivers.TaskConfig{
+		Name:    taskName,
+		AllocID: allocID,
+	}
 
 	//TODO do we need to se this accurately? Or will RecoverTask handle it?
 	ls.TaskHandle.State = drivers.TaskStateUnknown
 
-	//TODO do we need an envelope so drivers know this is an old state?
-	ls.TaskHandle.SetDriverState(t.HandleID)
+	// A ReattachConfig to the pre09 executor is sent
+	var raw []byte
+	var handle taskRunnerHandle08
+	if err := json.Unmarshal([]byte(t.HandleID), &handle); err != nil {
+		fmt.Println("ERR: ", err)
```

Comment:
Go ahead and add an error return here (maybe wrap the error in something mentioning the old handle couldn't be restored).

We'll fail the allocation which is the most defensive thing to be done (until we have discoverable recovery which doesn't rely on state).

Reply:
Ah yeah you spotted some println debugging I forgot to remove :sweat_smile: